packages/react-core/src/components/Chip/__tests__/Chip.test.tsx

```diff
@@ -1,59 +1,234 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import { Chip } from '../Chip';
+import styles from '@patternfly/react-styles/css/components/Chip/chip';
 
-describe('Chip', () => {
-  test('overflow', () => {
-    const { asFragment } = render(
-      <Chip className="my-chp-cls" isOverflowChip>
-        4 more
+['default chip', 'overflow chip'].forEach((chipType) => {
+  const isOverflowChip = chipType === 'overflow chip';
+
+  test(`Renders without children for ${chipType}`, () => {
+    render(
+      <div data-testid="container">
+        <Chip isOverflowChip={isOverflowChip} />
+      </div>
+    );
+
+    expect(screen.getByTestId('container').firstChild).toBeVisible();
+  });
+
+  test(`Renders with class ${styles.chip} on the ${chipType} container element`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
+      </Chip>
+    );
+
+    expect(screen.getByTestId('container')).toHaveClass(styles.chip);
+  });
+
+  test(`Renders with class ${styles.chipContent} around the ${chipType} content`, () => {
+    render(
+      <Chip isOverflowChip={isOverflowChip} data-testid="container">
+        Chip text
       </Chip>
     );
-    expect(asFragment()).toMatchSnapshot();
+
+    expect(screen.getByTestId('container').firstChild).toHaveClass(styles.chipContent);
+  });
+
+  test(`Renders ${chipType} children with class ${styles.chipText}`, () => {
+    render(<Chip isOverflowChip={isOverflowChip}>Chip text</Chip>);
+
+    expect(screen.getByText('Chip text')).toHaveClass(styles.chipText);
+  });
```

Comment:
I added a comment for the first test here, since the overflow chip won't have exactly the one class (it'll also have pf-m-overlfow, which we test elsewhere). Let me know what you think when I push the updates